File: app/src/main/java/com/arctouch/codechallenge/ui/activity/presenter/HomeActivityPresenter.java
package com.arctouch.codechallenge.ui.activity.presenter;

import com.arctouch.codechallenge.entity.Movie;

import java.util.List;

import io.reactivex.disposables.CompositeDisposable;

public interface HomeActivityPresenter {

    void getUpcomingMovies();
    void getMoreUpcomingMovies();
    void start();
    void getSearchMovies(String query);
    void getMoreSearchMovies(String query);
    void clearMoviesList();

    void clearCompositeDisposables();

    interface View {
        void refreshMoviesList(List<Movie> movies);
        void addMoviesToList(List<Movie> movies);
        void goToListTop();
    }
}

File: app/src/main/java/com/arctouch/codechallenge/ui/activity/presenter/HomeActivityPresenter.kt
package com.arctouch.codechallenge.ui.activity.presenter

import com.arctouch.codechallenge.entity.Movie

interface HomeActivityPresenter {

    fun getUpcomingMovies()
    fun getMoreUpcomingMovies()
    fun start()
    fun getSearchMovies(query: String)
    fun getMoreSearchMovies(query: String)
    fun clearMoviesList()

    fun clearCompositeDisposables()

    interface View {
        fun refreshMoviesList(movies: List<Movie>)
        fun addMoviesToList(movies: List<Movie>)
        fun goToListTop()
    }
}
